### PR TITLE
LWRP Terrain detail mesh shader

### PR DESCRIPTION
**Pull request by Verasl**
### Purpose of this PR
We are currently missing a solution for detail vertex lit meshes in the terrain system, as users cannot easily override this, this PR adds a basic vertex lit shadow receiving shader that overrides the builtin one.

DetailMeshShader

---
### Release Notes
-LWRP compatible Vertex Lit shader for detail meshes on the terrain system

---
### Testing status
**Katana Tests**: All Green

**Manual Tests**: Used shader in boat demo to add many small rocks around, it looks good and works with light mapping.

**Automated Tests**: Will add a more comprehensive terrain test after more terrain features are added(speedtree shader, etc)

---
### Overall Product Risks
**Technical Risk**: Low - Overrides a builtin shader that already didn't work with LWRP

**Halo Effect**: None - Doesn't touch any existing files

---
### Comments to reviewers
Check shader naming, I've tried to follow whats laid out in Core Common.hlsl for what makes sense. This will need updating once the shader API changes land for LWRP.

### CODE REVIEW

**Comment by Verasl** [comment]
Green on Katana Tests

File: com.unity.render-pipelines.lightweight/LWRP/Shaders/Terrain/VertexLit.shader [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    
+            Varyings Vert(Attributes input)
+            {
+                Varyings output = (Varyings)0;
+                UNITY_SETUP_INSTANCE_ID(input);
+                UNITY_INITIALIZE_VERTEX_OUTPUT_STEREO(output);
+                
+                // Vertex attributes
+                output.UV0Lightmap.xy = TRANSFORM_TEX(input.UV0, _MainTex);
+                output.UV0Lightmap.zw = input.UV1.xy * unity_LightmapST.xy + unity_LightmapST.zw;
```
Comment (phi-lira) [inline com.unity.render-pipelines.lightweight/LWRP/Shaders/Terrain/VertexLit.shader]: Some mobile GPUs will consider this packing a dependent texture fetch. It's not always better so pack uvs especially when you have interpolator space. I

File: com.unity.render-pipelines.lightweight/LWRP/Shaders/Terrain/VertexLit.shader [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                output.LightingFog.w = ComputeFogFactor(output.PositionCS.z);
+                
+                return output;
+            }
+    
+            TEXTURE2D(_MainTex);       SAMPLER(sampler_MainTex);
+    
+            half4 Frag(Varyings input) : SV_Target
+            {
+                half3 bakedGI = SampleLightmap(input.UV0Lightmap.zw, half3(0.0, 1.0, 0.0));
```
Comment (phi-lira) [inline com.unity.render-pipelines.lightweight/LWRP/Shaders/Terrain/VertexLit.shader]: If you don't plan to support directional lightmaps you can remove the DIRLIGHTMAP_COMBINED keyword from the shader.

File: com.unity.render-pipelines.lightweight/LWRP/Shaders/Terrain/VertexLit.shader (hunk 1 above)
Comment (Verasl) [reply com.unity.render-pipelines.lightweight/LWRP/Shaders/Terrain/VertexLit.shader]: Ah this explains why it was separate flat2's in the first place, good to know. Will change back to two separate ones.

File: com.unity.render-pipelines.lightweight/LWRP/Shaders/Terrain/VertexLit.shader (hunk 2 above)
Comment (Verasl) [reply com.unity.render-pipelines.lightweight/LWRP/Shaders/Terrain/VertexLit.shader]: Not in this iteration, will remove

File: com.unity.render-pipelines.lightweight/LWRP/Shaders/Terrain/VertexLit.shader [hunk 3]
```diff
@@ -56,10 +55,11 @@ Shader "Hidden/TerrainEngine/Details/Vertexlit"
     
             struct Varyings
             {
-                float4  UV0Lightmap     : TEXCOORD0; // UV0, Lightmap UVs
-                half4   Color           : TEXCOORD1; // Vertex Color
-                half4   LightingFog     : TEXCOORD2; // Vetex Lighting, Fog Factor
-                float4  ShadowCoords    : TEXCOORD3; // Shadow UVs
+                float2  UV01            : TEXCOORD0; // UV0
```
Comment (phi-lira) [inline com.unity.render-pipelines.lightweight/LWRP/Shaders/Terrain/VertexLit.shader]: minor things. this should probably just be uv0.

**Pull request merged**